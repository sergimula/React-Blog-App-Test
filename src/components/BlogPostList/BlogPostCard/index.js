import React from 'react'
import {
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  Grid,
  makeStyles,
  Typography,
} from '@material-ui/core'
import PropTypes from 'prop-types'

import styles from './styles'

const useStyles = makeStyles(styles)

const BlogPostCard = (props) => {
  const classes = useStyles()

  return (
    <Card className={classes.root}>
      <CardActionArea>
        <Grid container>
          <Grid item xs={6}>
            <CardMedia
              component='img'
              alt={props.title}
              height='140'
              image={props.imageUrl}
              title={props.title}
            />
          </Grid>
          <Grid item xs={6}>
            <CardContent>
              <Typography gutterBottom variant='h5' component='h2'>
                {props.title}
              </Typography>
              <Typography variant='body2' color='textSecondary' component='p'>
                {props.description}
              </Typography>
            </CardContent>
          </Grid>
        </Grid>
      </CardActionArea>
    </Card>
  )
}

BlogPostCard.propTypes = {
  description: PropTypes.string,
  imageUrl: PropTypes.string,
  title: PropTypes.string,
}

export default BlogPostCard
