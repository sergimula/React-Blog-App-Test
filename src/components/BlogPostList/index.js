import React from 'react'
import { Grid } from '@material-ui/core'
import PropTypes from 'prop-types'

import BlogPostCard from './BlogPostCard'

const BlogPostList = (props) => (
  <Grid alignContent='center' container direction='column' justify='center'>
    {props.posts.map((post) => (
      <BlogPostCard
        description={post.description}
        imageUrl={post.image}
        key={post.id}
        title={post.title}
      />
    ))}
  </Grid>
)

BlogPostList.propTypes = {
  posts: PropTypes.array,
}

export default BlogPostList
