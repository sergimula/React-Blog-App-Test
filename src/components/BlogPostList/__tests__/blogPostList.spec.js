import React from 'react'
import renderer from 'react-test-renderer'

import BlogPostList from '../index'
import BlogPostCardStyles from '../BlogPostCard/styles'
import * as stubs from './stubs'

describe('BlogPostList', () => {
  it('should match BlogPostList', () => {
    const wrapper = renderer.create(<BlogPostList posts={stubs.posts} />)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('styles', () => {
  describe('BlogPostCard', () => {
    expect(BlogPostCardStyles).toMatchSnapshot()
  })
})
