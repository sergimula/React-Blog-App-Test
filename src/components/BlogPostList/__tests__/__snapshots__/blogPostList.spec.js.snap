// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `[Function]`;

exports[`BlogPostList should match BlogPostList 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <img
            alt="React and Next are awesome"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="140"
            src="https://picsum.photos/id/0/200/200"
            title="React and Next are awesome"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              React and Next are awesome
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Est ne harum postulant, mei denique assueverit ei. An malorum eripuit vim, mei te voluptua ponderum. Eu sed eirmod antiopam argumentum. An natum assum dolorem sea.
            </p>
          </div>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <img
            alt="Vintage motorcycles are awesome"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="140"
            src="https://picsum.photos/id/278/200/200"
            title="Vintage motorcycles are awesome"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Vintage motorcycles are awesome
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Est ne harum postulant, mei denique assueverit ei. An malorum eripuit vim, mei te voluptua ponderum. Eu sed eirmod antiopam argumentum. An natum assum dolorem sea.
            </p>
          </div>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <img
            alt="Love Munchking, very funny card game"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="140"
            src="https://picsum.photos/id/307/200/200"
            title="Love Munchking, very funny card game"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Love Munchking, very funny card game
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Est ne harum postulant, mei denique assueverit ei. An malorum eripuit vim, mei te voluptua ponderum. Eu sed eirmod antiopam argumentum. An natum assum dolorem sea.
            </p>
          </div>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <img
            alt="Covid-19 hits again"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="140"
            src="https://picsum.photos/id/338/200/200"
            title="Covid-19 hits again"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Covid-19 hits again
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Est ne harum postulant, mei denique assueverit ei. An malorum eripuit vim, mei te voluptua ponderum. Eu sed eirmod antiopam argumentum. An natum assum dolorem sea.
            </p>
          </div>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <img
            alt="React and Next are awesome"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="140"
            src="https://picsum.photos/id/0/200/200"
            title="React and Next are awesome"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              React and Next are awesome
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Est ne harum postulant, mei denique assueverit ei. An malorum eripuit vim, mei te voluptua ponderum. Eu sed eirmod antiopam argumentum. An natum assum dolorem sea.
            </p>
          </div>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <img
            alt="Vintage motorcycles are awesome"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="140"
            src="https://picsum.photos/id/278/200/200"
            title="Vintage motorcycles are awesome"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Vintage motorcycles are awesome
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Est ne harum postulant, mei denique assueverit ei. An malorum eripuit vim, mei te voluptua ponderum. Eu sed eirmod antiopam argumentum. An natum assum dolorem sea.
            </p>
          </div>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <img
            alt="Love Munchking, very funny card game"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="140"
            src="https://picsum.photos/id/307/200/200"
            title="Love Munchking, very funny card game"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Love Munchking, very funny card game
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Est ne harum postulant, mei denique assueverit ei. An malorum eripuit vim, mei te voluptua ponderum. Eu sed eirmod antiopam argumentum. An natum assum dolorem sea.
            </p>
          </div>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <img
            alt="Covid-19 hits again"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="140"
            src="https://picsum.photos/id/338/200/200"
            title="Covid-19 hits again"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              Covid-19 hits again
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              Est ne harum postulant, mei denique assueverit ei. An malorum eripuit vim, mei te voluptua ponderum. Eu sed eirmod antiopam argumentum. An natum assum dolorem sea.
            </p>
          </div>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</div>
`;
